backend/services/implementations/testSessionService.ts

```diff
@@ -38,6 +38,53 @@ class TestSessionService implements ITestSessionService {
       startTime: newTestSession.startTime,
     };
   }
+
+  async getAllTestSessions(): Promise<Array<TestSessionResponseDTO>> {
+    let testSessionDtos: Array<TestSessionResponseDTO> = [];
+
+    try {
+      const testSessions: Array<TestSession> = await MgTestSession.find();
+      testSessionDtos = await this.mapTestSessionsToTestSessionDTOs(
+        testSessions,
+      );
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to get test sessions. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+
+    return testSessionDtos;
+  }
+
+  private async mapTestSessionsToTestSessionDTOs(
```

Comment:
not sure if you know the answer to this - but how does mongo handle large requests? i.e say we have thousands of test sessions, if we make a request for "all test sessions", how does it behave?

Reply:
think it should be fine - would probably become a problem if we were querying millions of documents

Reply:
cool, sounds good